lib/getPostcssResult.js

```diff
@@ -19,73 +19,65 @@ const postcssProcessor = postcss();
  *
  * @returns {Promise<Result>}
  */
-module.exports = function getPostcssResult(stylelint, options = {}) {
+module.exports = async function getPostcssResult(stylelint, options = {}) {
 	const cached = options.filePath ? stylelint._postcssResultCache.get(options.filePath) : undefined;
 
-	if (cached) return Promise.resolve(cached);
+	if (cached) {
+		return cached;
+	}
+
+	if (stylelint._options.syntax) {
+		return Promise.reject(
```

Comment:
When `throw` and error in async function it is converted to rejected Promise. So for being more explicit I converted all `throw` into `return Promise.reject()` through this PR.